### PR TITLE
Use correct registries to integrate with new packages workflows

### PR DESCRIPTION
**Pull request by abhay-krishna**
Use correct registries to integrate with new packages workflows.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/curatedpackages/regional_registry.go [hunk 1]
```diff
@@ -101,10 +101,10 @@ func TestRegistryWithAuthToken(authToken, registry string, do Do) error {
 
 // GetRegionalRegistry get the regional registry corresponding to defaultRegistry in a specific region.
 func GetRegionalRegistry(defaultRegistry, region string) string {
-	if strings.Contains(defaultRegistry, devAccount) {
+	if strings.Contains(defaultRegistry, devRegistry) {
```
Comment (pokearu) [inline pkg/curatedpackages/regional_registry.go:104]: For this do we need to change the bundle to have the dev registry??

**Comment by mitalipaygude** [comment]
Can we add unit tests for these changes? also a section in the description that describes how these changes were tested in general?

File: pkg/curatedpackages/reader.go [hunk 2]
```diff
@@ -17,14 +17,14 @@ import (
...
-	publicProdECR     = "public.ecr.aws/" + prodAccount
-	publicDevECR      = "public.ecr.aws/" + devAccount
-	publicStagingECR  = "public.ecr.aws/" + stagingAccount
+	prodRegistry      = "eks-anywhere"
+	devRegistry       = "x3k6m8v0"
+	stagingRegistry   = "w9m0f3l5"
+	publicProdECR     = "public.ecr.aws/" + prodRegistry
+	publicDevECR      = "public.ecr.aws/" + devRegistry
+	publicStagingECR  = "public.ecr.aws/" + stagingRegistry
 	packageProdDomain = "783794618700.dkr.ecr.us-west-2.amazonaws.com"
```
Comment (sp1999) [inline pkg/curatedpackages/reader.go:26]:
```suggestion
	privateProdRegistry = "783794618700.dkr.ecr.us-west-2.amazonaws.com"
```

File: pkg/curatedpackages/reader.go [hunk 3]
```diff
@@ -17,14 +17,14 @@ import (
...
-	packageDevDomain  = "857151390494.dkr.ecr.us-west-2.amazonaws.com"
+	packageDevDomain  = "067575901363.dkr.ecr.us-west-2.amazonaws.com"
```
Comment (sp1999) [inline pkg/curatedpackages/reader.go:27]:
```suggestion
	privateDevRegistry  = "067575901363.dkr.ecr.us-west-2.amazonaws.com"
```

File: pkg/curatedpackages/regional_registry.go [hunk 4]
```diff
@@ -17,7 +17,7 @@ import (
 const (
 	devRegionalECR       string = "067575901363.dkr.ecr.us-west-2.amazonaws.com"
 	devRegionalPublicECR string = "public.ecr.aws/x3k6m8v0"
-	stagingRegionalECR   string = "067575901363.dkr.ecr.us-west-2.amazonaws.com"
```
Comment (sp1999) [inline pkg/curatedpackages/regional_registry.go:20]: Why don't we add a `stagingRegionalECRMap` to map all regions to the corresponding staging account private ECR similar to `prodRegionalECRMap`?

File: pkg/curatedpackages/regional_registry.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ import (
...
+	stagingRegionalECR   string = "724423470321.dkr.ecr.us-west-2.amazonaws.com"
```
Comment (sp1999) [inline pkg/curatedpackages/regional_registry.go:20]: Why don't we add a stagingRegionalECRMap to map all regions to the corresponding staging account private ECR similar to prodRegionalECRMap?

Comment (pokearu) [reply pkg/curatedpackages/regional_registry.go:20]: We are not going to run staging tests in every region, we just need one place where we can get staging artifacts from. So PDX gamma would do.

File: pkg/curatedpackages/regional_registry.go (hunk 1 above)
Comment (abhay-krishna) [reply pkg/curatedpackages/regional_registry.go:104]: ah yeah I think I need to have two separate vars for the old/new dev registries.

File: pkg/curatedpackages/packagecontrollerclient.go [hunk 6]
```diff
@@ -241,26 +241,26 @@ func (pc *PackageControllerClient) Enable(ctx context.Context) error {
...
+	defaultImageRegistry = prodNonRegionalPrivateRegistryURI
+	registry := prodPublicRegistryAlias
+	if strings.Contains(pc.chart.Image(), devNonRegionalPublicRegistryAlias) {
+		registry = devRegionalPublicRegistryAlias
+		defaultImageRegistry = devRegionalPrivateRegistryURI
+		sourceRegistry = devRegionalPublicRegistryURI
+	}
+	if strings.Contains(pc.chart.Image(), stagingPublicRegistryAlias) {
+		registry = stagingPublicRegistryAlias
+		defaultImageRegistry = devRegionalPrivateRegistryURI
```
Comment (vivek-koppuru) [inline pkg/curatedpackages/packagecontrollerclient.go:254]: Is this not going to be the gamma account?

File: pkg/curatedpackages/regional_registry.go [hunk 7]
```diff
@@ -101,13 +65,13 @@ func TestRegistryWithAuthToken(authToken, registry string, do Do) error {
 
 // GetRegionalRegistry get the regional registry corresponding to defaultRegistry in a specific region.
 func GetRegionalRegistry(defaultRegistry, region string) string {
-	if strings.Contains(defaultRegistry, devAccount) {
-		return devRegionalECR
+	if strings.Contains(defaultRegistry, devNonRegionalPublicRegistryAlias) {
+		return devRegionalPrivateRegistryURI
```
Comment (sp1999) [inline pkg/curatedpackages/regional_registry.go:69]: nit: add unit test to cover this line as well

File: pkg/curatedpackages/registry_constants.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
+package curatedpackages
+
+const (
+	devNonRegionalPublicRegistryAlias = "l0g8r8j6"
```
Comment (pokearu) [inline pkg/curatedpackages/registry_constants.go:4]: can we not use the same one as `devRegionalPublicRegistryAlias`?

**Review by pokearu — APPROVED** [review]
/lgtm
/approve

**Pull request merged**